socket_linux_amd.go

```diff
@@ -0,0 +1,134 @@
+//go:build android && amd64
+// +build android,amd64
+
+package tcp
+
+import (
+	"fmt"
+	"os"
+	"time"
+
+	"golang.org/x/sys/unix"
+)
+
+const maxEpollEvents = 32
+
+// createSocket creates a socket with necessary options set.
+func createSocketZeroLinger(zeroLinger bool) (fd int, err error) {
+	// Create socket
+	fd, err = _createNonBlockingSocket()
+	if err == nil {
+		if zeroLinger {
```

Comment:
you can put the conditions on one line
`if err == nil && zeroLinger`

Reply:
Thanks! much neater now